packages/api-client/index.ts

```diff
@@ -0,0 +1,229 @@
+import Axios, { AxiosInstance } from "axios";
+import {
+  CreatePlanDto,
+  CreateStudentDto,
+  GetPlanResponse,
+  GetStudentResponse,
+  INEUAndPrereq,
+  INEUOrPrereq,
+  LoginStudentDto,
+  ScheduleCourse,
+  UpdatePlanDto,
+  UpdatePlanResponse,
+  UpdateStudentDto,
+  UpdateStudentResponse,
+} from "@graduate/common";
+import { plainToInstance } from "class-transformer";
+
+class APIClient {
+  private axios: AxiosInstance;
+
+  auth = {
+    login: async (
+      loginUserDto: LoginStudentDto
+    ): Promise<GetStudentResponse> => {
+      const data = (await this.axios.post("/auth/login", { ...loginUserDto }))
+        .data;
+      return plainToInstance(GetStudentResponse, data);
+    },
+    register: async (
+      createStudentDto: CreateStudentDto
+    ): Promise<GetStudentResponse> => {
+      const data = (
+        await this.axios.post("/auth/register", { ...createStudentDto })
+      ).data;
+      return plainToInstance(GetStudentResponse, data);
+    },
+  };
+
+  student = {
+    update: async (
+      updateStudentDto: UpdateStudentDto,
+      jwt: string
+    ): Promise<UpdateStudentResponse> => {
+      const data = (
+        await this.axios.patch(
+          "/students/me",
+          { ...updateStudentDto },
+          { headers: { Authorization: `Bearer ${jwt}` } }
```

Comment:
We pass the JWT into the header because the current `api-v2` doesn't support JWT via cookies. I talked to @aryanshah701 about this and he mentioned that for testing purposes, we should align the api-client to the current api-v2. Then once we make the ticket to switch the backend to utilise cookies, we will also change this accordingly